### DISCUSSION TITLE
Issue with PC increment

### DISCUSSION THREAD

**Discussion by BLACKFirework** [NUS-CG3207/labs#15] [Lab 2]
can't find any flags that can be used to let ALU do PC related calculation

a bit tempted to just write PC_IN = PC + 4 in RV but I know it's structurally bad

can we add flags or modify the i/o for ALU so that in certain conditions we can let PC_IN = ALUResult?

Or any hints for alternatives :")))))

**Comment by r-panicker** [comment]
It is ok to use a separate '+' for PC_IN calculation. The design instructions allow for it, as it mentions that all operations on 32-bit values other than PC+ calculation should be done in the ALU.
It can't be done in ALU for a single-cycle processor as you need ALU for comparison in the same cycle while executing the branch. Using ALU for PC incrementation is possible only in a multi-cycle design where one instruction takes more than one cycle to complete.
Using '+' is actually recommended if you wish to infer an adder, as it allows the synthesis tool to make use of the built-in support FPGAs have for dealing with carry efficiently.
